Improve safelinks pattern

jan-kiszka opened the pull request:
This also ensure that the complete link is replaced when citing a
mangled email.

jan-kiszka committed:
Improve regexp for catching mangled URLs

This ensures that the complete link is replaced when citing a mangled
email in plaintext.

Signed-off-by: Jan Kiszka <jan.kiszka@web.de>

jan-kiszka commented:
BTW, how about releasing a new version that contains link-unmangling for replies?

cphyc commented on src/decoders.js:
```diff
@@ -1,4 +1,4 @@
-let outlook_regex = /https:\/\/\w+\.safelinks\.protection\.outlook\.com\/\?url=(\S+?)((?:\&amp;data.*)+$|$)/i;
+let outlook_regex = /https:\/\/\w+\.safelinks\.protection\.outlook\.com\/\?url=([^&]*)(&[-+a-zA-Z0-9%&/=.;]*)?/i;
```
Shouldn't the `+`, `-` and `.` be escaped above?
```suggestion
let outlook_regex = /https:\/\/\w+\.safelinks\.protection\.outlook\.com\/\?url=([^&]*)(&[\-\+a-zA-Z0-9%&/=\.;]*)?/i;
```

jan-kiszka replied:
Don't ask me when exactly they need to be escaped. This seems to work fine. But there is more to improve, e.g. excluding trailing `.`.

cphyc reviewed (approved):
Thanks for the PR!
Would it be possible to add a test for this case?

Once merged, I'll happily create a new release.

jan-kiszka committed:
Add test case for text replacement

Signed-off-by: Jan Kiszka <jan.kiszka@web.de>

jan-kiszka commented:
Will have a look into test cases.

jan-kiszka committed:
Add another decoder test pattern for safelinks

This contains some domain name in the data field, checking that the
decoder accepts '.' in the middle of the URL.

Signed-off-by: Jan Kiszka <jan.kiszka@web.de>

jan-kiszka commented:
Further tuned the regexps and also added some tests. Seems to work fine here so far.

cphyc commented:
Thanks for the PR! I also created a new release that includes your code. You should be able to update from TB directly.
